### DISCUSSION TITLE
Error Loading Custom Robot Model in MoveIt2

### DISCUSSION THREAD

**Discussion by robot-RL-AI** [moveit/moveit2#2299] [Q&A] [answered]
I am using ROS2 Humble. I'm attempting to set up my personally crafted robotic manipulator to utilize MoveIt2 for motion planning and simulate its actions within Gazebo. However, I'm encountering a number of quandaries while endeavoring to execute the demo.launch.py script. To be more precise, I'm being confronted with a series of error messages that seem to pertain to the controller manager and the loading of the robot model.

Here are the specifics: ROS Distro: ROS2 Humble Packages: gazebo_ros2_control, moveit2, and a tailor-made package for my individual robot.

Error message:

Using load_yaml() directly is deprecated. Use xacro.load_yaml() instead. [static_transform_publisher-1] translation: ('0.000000', '0.000000', '0.000000') [static_transform_publisher-1] rotation: ('0.000000', '0.000000', '0.000000', '1.000000') [static_transform_publisher-1] from 'world' to 'base_link' [rviz2-4] Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway. [ros2_control_node-5] terminate called after throwing an instance of 'pluginlib::LibraryLoadException' [ros2_control_node-5] what(): According to the loaded plugin descriptions the class gazebo_ros2_control/GazeboSystem with base class type hardware_interface::SystemInterface does not exist. Declared types are fake_components/GenericSystem mock_components/GenericSystem test_hardware_components/TestSystemCommandModes test_hardware_components/TestTwoJointSystem test_system [ros2_control_node-5] Stack trace (most recent call last): [ros2_control_node-5] #16 Object "", at 0xffffffffffffffff, in [ros2_control_node-5] #15 Object "/opt/ros/humble/lib/controller_manager/ros2_control_node", at 0x55d431a4dd84, in [ros2_control_node-5] #14 Source "../csu/libc-start.c", line 392, in __libc_start_main_impl [0x7f701de29e3f] [ros2_control_node-5] #13 Source "../sysdeps/nptl/libc_start_call_main.h", line 58, in __libc_start_call_main [0x7f701de29d8f] [ros2_control_node-5] #12 Object "/opt/ros/humble/lib/controller_manager/ros2_control_node", at 0x55d431a4d89e, in [ros2_control_node-5] #11 Object "/opt/ros/humble/lib/libcontroller_manager.so", at 0x7f701e7b9c27, in controller_manager::ControllerManager::ControllerManager(std::shared_ptr<rclcpp::executor>, std::__cxx11::basic_string<char, std::char_traits<char="">, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char="">, std::allocator<char> > const&, rclcpp::NodeOptions const&) [ros2_control_node-5] #10 Object "/opt/ros/humble/lib/libcontroller_manager.so", at 0x7f701e7b8dd7, in controller_manager::ControllerManager::init_resource_manager(std::__cxx11::basic_string<char, std::char_traits<char="">, std::allocator<char> > const&) [ros2_control_node-5] #9 Object "/opt/ros/humble/lib/libhardware_interface.so", at 0x7f701e1c2207, in hardware_interface::ResourceManager::load_urdf(std::__cxx11::basic_string<char, std::char_traits<char="">, std::allocator<char> > const&, bool) [ros2_control_node-5] #8 Object "/opt/ros/humble/lib/libhardware_interface.so", at 0x7f701e1ca856, in [ros2_control_node-5] #7 Object "/opt/ros/humble/lib/libhardware_interface.so", at 0x7f701e1a1858, in [ros2_control_node-5] #6 Object "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.30", at 0x7f701e2ae517, in __cxa_throw [ros2_control_node-5] #5 Object "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.30", at 0x7f701e2ae2b6, in std::terminate() [ros2_control_node-5] #4 Object "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.30", at 0x7f701e2ae24b, in [ros2_control_node-5] #3 Object "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.30", at 0x7f701e2a2bbd, in [ros2_control_node-5] #2 Source "./stdlib/abort.c", line 79, in abort [0x7f701de287f2] [ros2_control_node-5] #1 Source "../sysdeps/posix/raise.c", line 26, in raise [0x7f701de42475] [ros2_control_node-5] #0 | Source "./nptl/pthread_kill.c", line 89, in __pthread_kill_internal [ros2_control_node-5] | Source "./nptl/pthread_kill.c", line 78, in __pthread_kill_implementation [ros2_control_node-5] Source "./nptl/pthread_kill.c", line 44, in __pthread_kill [0x7f701de96a7c] [ros2_control_node-5] Aborted (Signal sent by tkill() 29670 1000) [move_group-3] terminate called after throwing an instance of 'rclcpp::exceptions::InvalidParameterTypeException' [move_group-3] what(): parameter 'robot_description_planning.joint_limits.joint_2.max_velocity' has invalid type: expected [double] got [integer] [move_group-3] Stack trace (most recent call last): [move_group-3] #18 Object "", at 0xffffffffffffffff, in [move_group-3] #17 Object "/opt/ros/humble/lib/moveit_ros_move_group/move_group", at 0x556b13d91784, in [move_group-3] #16 Source "../csu/libc-start.c", line 392, in __libc_start_main_impl [0x7effd4229e3f] [move_group-3] #15 Source "../sysdeps/nptl/libc_start_call_main.h", line 58, in __libc_start_call_main [0x7effd4229d8f] [move_group-3] #14 Object "/opt/ros/humble/lib/moveit_ros_move_group/move_group", at 0x556b13d90322, in [move_group-3] #13 Object "/opt/ros/humble/lib/libmoveit_cpp.so.2.5.4", at 0x7effd4d78094, in moveit_cpp::MoveItCpp::MoveItCpp(std::shared_ptr<rclcpp::node> const&, moveit_cpp::MoveItCpp::Options const&) [move_group-3] #12 Object "/opt/ros/humble/lib/libmoveit_cpp.so.2.5.4", at 0x7effd4d75a3b, in moveit_cpp::MoveItCpp::loadPlanningSceneMonitor(moveit_cpp::MoveItCpp::PlanningSceneMonitorOptions const&) [move_group-3] #11 Object "/opt/ros/humble/lib/libmoveit_planning_scene_monitor.so.2.5.4", at 0x7effd4c9960a, in planning_scene_monitor::PlanningSceneMonitor::PlanningSceneMonitor(std::shared_ptr<rclcpp::node> const&, std::__cxx11::basic_string<char, std::char_traits<char="">, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char="">, std::allocator<char> > const&) [move_group-3] #10 Object "/opt/ros/humble/lib/libmoveit_planning_scene_monitor.so.2.5.4", at 0x7effd4c9955a, in planning_scene_monitor::PlanningSceneMonitor::PlanningSceneMonitor(std::shared_ptr<rclcpp::node> const&, std::shared_ptr<planning_scene::planningscene> const&, std::__cxx11::basic_string<char, std::char_traits<char="">, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char="">, std::allocator<char> > const&) [move_group-3] #9 Object "/opt/ros/humble/lib/libmoveit_robot_model_loader.so.2.5.4", at 0x7effd4443c1e, in robot_model_loader::RobotModelLoader::RobotModelLoader(std::shared_ptr<rclcpp::node> const&, std::__cxx11::basic_string<char, std::char_traits<char="">, std::allocator<char> > const&, bool) [move_group-3] #8 Object "/opt/ros/humble/lib/libmoveit_robot_model_loader.so.2.5.4", at 0x7effd44423d9, in robot_model_loader::RobotModelLoader::configure(robot_model_loader::RobotModelLoader::Options const&) [move_group-3] #7 Object "/opt/ros/humble/lib/libmoveit_robot_model_loader.so.2.5.4", at 0x7effd443c1a2, in [move_group-3] #6 Object "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.30", at 0x7effd46ae517, in __cxa_throw [move_group-3] #5 Object "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.30", at 0x7effd46ae2b6, in std::terminate() [move_group-3] #4 Object "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.30", at 0x7effd46ae24b, in [move_group-3] #3 Object "/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.30", at 0x7effd46a2bbd, in [move_group-3] #2 Source "./stdlib/abort.c", line 79, in abort [0x7effd42287f2] [move_group-3] #1 Source "../sysdeps/posix/raise.c", line 26, in raise [0x7effd4242475] [move_group-3] #0 | Source "./nptl/pthread_kill.c", line 89, in __pthread_kill_internal [move_group-3] | Source "./nptl/pthread_kill.c", line 78, in __pthread_kill_implementation [move_group-3] Source "./nptl/pthread_kill.c", line 44, in __pthread_kill [0x7effd4296a7c] [move_group-3] Aborted (Signal sent by tkill() 29666 1000) [rviz2-4] Warning: class_loader.impl: SEVERE WARNING!!! A namespace collision has occurred with plugin factory for class rviz_default_plugins::displays::InteractiveMarkerDisplay. New factory will OVERWRITE existing one. This situation occurs when libraries containing plugins are directly linked against an executable (the one running right now generating this message). Please separate plugins out into their own library or just don't link against the library and use either class_loader::ClassLoader/MultiLibraryClassLoader to open. [rviz2-4] at line 253 in /opt/ros/humble/include/class_loader/class_loader/class_loader_core.hpp [rviz2-4] Error: Could not parse the SRDF XML File. Error=XML_ERROR_EMPTY_DOCUMENT ErrorID=13 (0xd) Line number=0 [rviz2-4] at line 715 in ./src/model.cpp ^C[rviz2-4] [INFO] [1684922701.918329435] [rclcpp]: signal_handler(signum=2)

**Comment by bmaxdk** [comment, marked as answer]
Your error indicated below:
`[move_group-3] terminate called after throwing an instance of 'rclcpp::exceptions::InvalidParameterTypeException' [move_group-3] what(): parameter 'robot_description_planning.joint_limits.joint_2.max_velocity' has invalid type: expected [double] got [integer]`

You may want to check error message where it shows `invalid Parameter Type Exception (move_group-3)`. 
This error message is related to an incorrect data type for the `robot_description_planning.joint_limits.joint_2.max_velocity` parameter. The system expects a **double**, but an **integer** was found. This can usually be fixed by editing the corresponding configuration file and ensuring your floating-point number such as `1.0` instead of `1`

If you still meet same issue, double check your `URDF` or `configuration files`. Since `gazebo_ros2_control/GazeboSystem` class has loading error, where the error message says that this class does not exist and it only lists the declared types. This mismatch between the plugin declared in your `URDF` or `YAML` files need to follow the correct plugins.
`[ros2_control_node-5] terminate called after throwing an instance of 'pluginlib::LibraryLoadException' [ros2_control_node-5] what(): According to the loaded plugin descriptions the class gazebo_ros2_control/GazeboSystem with base class type hardware_interface::SystemInterface does not exist. Declared types are fake_components/GenericSystem mock_components/GenericSystem test_hardware_components/TestSystemCommandModes test_hardware_components/TestTwoJointSystem test_system`

**Reply by robot-RL-AI** [reply]
Thank you it was type issue
